### PR TITLE
add vernemq sidecar

### PR DESCRIPTION
**Pull request by andersonluisribeiro**
* **Please check if the PR fulfills these requirements**
  * [ ] Tests for the changes have been added (for bug fixes / features)
  * [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add VerneMQ, V2K and K2V sidecar

* **What is the current behavior?** (You can also link to an open issue here)
Therie is no sidecar

* **What is the new behavior (if this is a feature change)?**
Sidecar for VerneMQ, V2K and K2V

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
Yes, this PR is connected to dojot/dojot#1802

* **Other information**:
For this to work, this PR needs to be merged and a new operator image needs to be generated: https://github.com/dojot/vmq-operator/pull/2 . The new operator image should have the same dojot tag. The "dojot_version" is used as image tag.

### CODE REVIEW

**Review by marianoleonardo — DISMISSED** [review]
LGTM!

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
...
...D0gb24KcGx1Z2lucy5kb2pvdF9kaXNjb25uZWN0X3BsdWdpbiA9IG9uCnBsdWdpbnMudm1xX2s4cy5wcmlvcml0eSA9IDMKcGx1Z2lucy5kb2pvdF9hY2xfcGx1Z2luLnByaW9yaXR5ID0gMgpwbHVnaW5zLmRvam90X2Rpc2Nvbm5lY3RfcGx1Z2luLnByaW9yaXR5ID0gMgpsb2cuY29uc29sZSA9IGNvbnNvbGUKbG9nLmNvbnNvbGUubGV2ZWwgPSBpbmZvCmxpc3RlbmVyLnNzbC5kZWZhdWx0ID0gMC4wLjAuMDo4ODgzCmxpc3RlbmVyLnNzbC5kZWZhdWx0LmNhZmlsZSA9ICAvdmVybmVtcS9jZXJ0L2NhLmNydApsaXN0ZW5lci5zc2wuZGVmYXVsdC5jZXJ0ZmlsZSA9IC92ZXJuZW1xL2NlcnQvaW90YWdlbnQtbXF0dC5jcnQKbGlzdGVuZXIuc3NsLmRlZmF1bHQua2V5ZmlsZSA9IC92ZXJuZW1xL2NlcnQvaW90YWdlbnQtbXF0dC5rZXkKbGlzdGVuZXIuc3NsLmRlZmF1bHQuY3JsZmlsZSA9ICAvdmVybmVtcS9jZXJ0L2NhLmNybApsaXN0ZW5lci5zc2wuZGVmYXVsdC51c2VfaWRlbnRpdHlfYXNfdXNlcm5hbWUgPSBvbgpsaXN0ZW5lci5zc2wuZGVmYXVsdC5yZXF1aXJlX2NlcnRpZmljYXRlID0gb24KbGlzdGVuZXIubWF4X2Nvbm5lY3Rpb25zID0gaW5maW5pdHkKZXJsYW5nLm1heF9wb3J0cyA9IDEzNDIxNzcyNwpzeXN0cmVlX2VuYWJsZWQgPSBvZmYKbGV2ZWxkYi5tYXhpbXVtX21lbW9yeS5wZXJjZW50ID0gMjAKbWF4X29ubGluZV9tZXNzYWdlcyA9IDUwMDAwCm1heF9vZmZsaW5lX21lc3NhZ2VzID0gNTAwMDAKbWF4X2luZmxpZ2h0X21lc3NhZ2VzID0gMTAwCnNoYXJlZF9zdWJzY3JpcHRpb25fcG9saWN5ID0gcHJlZmVyX2xvY2FsCmVybGFuZy5kaXN0cmlidXRpb25fYnVmZmVyX3NpemUgPSA1MTIwMDAKZXJsYW5nLmFzeW5jX3RocmVhZHMuc3RhY2tfc2l6ZSA9IDEzMTA3Mg=="
```
Comment (mprevide) [inline roles/vernemq/templates/z_vernemq_cluster.yaml.j2:30]: change in base 64 
`listener.ssl.default.cafile =  /vernemq/cert/ca.crt` to
`listener.ssl.default.cafile =  /vernemq/cert/cabundle.crt` to

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 [hunk 2]
```diff
@@ -45,6 +45,30 @@ spec:
...
       websocket: true
     plugins: []
+  sidecarBaseImage: dojot/cert-sidecar
+  sidecarEnv:
+    - name: CERT_SC_APP_SIDECAR_TO
+      value: "vernemq-k8s"
+    - name: CERT_SC_LOG_CONSOLE_LEVEL
+      value: "info"
+    - name: CERT_SC_CERTS_HOSTNAMES
+      value: "[\"vernemq-k8s\", \"localhost\"]"
```
Comment (mprevide) [inline roles/vernemq/templates/z_vernemq_cluster.yaml.j2:55]: Change `localhost` to `{{ dojot_domain_name }}`

**Review by mprevide — CHANGES_REQUESTED** [review]

File: roles/k2v-bridge/templates/k2v_bridge_cluster.yaml.j2 [hunk 3]
```diff
@@ -37,6 +37,69 @@ spec:
...
+        env:
+        - name: CERT_SC_APP_SIDECAR_TO
+          value: "k2v-bridge"
+        - name: CERT_SC_LOG_CONSOLE_LEVEL
+          value: "info"
+        - name: CERT_SC_CERTS_CRL
+          value: "false"  
+        - name: CERT_SC_CERTS_HOSTNAMES
+          value: "[\"k2v-bridge\"]"
+        - name: HOSTNAME
```
[the remark this answers is not in the document]
Comment (andersonluisribeiro) [reply roles/k2v-bridge/templates/k2v_bridge_cluster.yaml.j2]: Changed!

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 (hunk 1 above)
Comment (andersonluisribeiro) [reply roles/vernemq/templates/z_vernemq_cluster.yaml.j2]: Changed!

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 (hunk 2 above)
Comment (andersonluisribeiro) [reply roles/vernemq/templates/z_vernemq_cluster.yaml.j2]: Changed!

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 [hunk 4]
```diff
@@ -45,6 +45,30 @@ spec:
...
     plugins: []
+  sidecarBaseImage: dojot/cert-sidecar
+  sidecarEnv:
+    - name: CERT_SC_APP_SIDECAR_TO
+      value: "vernemq-k8s"
+    - name: CERT_SC_LOG_CONSOLE_LEVEL
+      value: "info"
+    - name: CERT_SC_CERTS_HOSTNAMES
+      value: "[\"vernemq-k8s\", \"{{ dojot_domain_name }}\"]"
+    - name: HOSTNAME
```
Comment (mprevide) [inline roles/vernemq/templates/z_vernemq_cluster.yaml.j2:56]: Remove HOSTNAME

File: roles/v2k-bridge/templates/v2k_bridge_cluster.yaml.j2 [hunk 5]
```diff
@@ -39,7 +39,67 @@ spec:
...
+        - name: CERT_SC_APP_SIDECAR_TO
+          value: "v2k-bridge"
+        - name: CERT_SC_LOG_CONSOLE_LEVEL
+          value: "info"
+        - name: CERT_SC_CERTS_CRL
+          value: "false"  
+        - name: CERT_SC_CERTS_HOSTNAMES
+          value: "[\"v2k-bridge\"]"
+        - name: HOSTNAME
+          value: "v2k-bridge"
```
Comment (mprevide) [inline roles/v2k-bridge/templates/v2k_bridge_cluster.yaml.j2:88]: Remove HOSTNAME

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 (hunk 4 above)
Comment (andersonluisribeiro) [reply roles/vernemq/templates/z_vernemq_cluster.yaml.j2]: Changed!

File: roles/v2k-bridge/templates/v2k_bridge_cluster.yaml.j2 (hunk 5 above)
Comment (andersonluisribeiro) [reply roles/v2k-bridge/templates/v2k_bridge_cluster.yaml.j2]: Changed!

**Review by mprevide — APPROVED** [review]
Great Job! :)

**Review by marianoleonardo — APPROVED** [review]
LGTM!

**Pull request merged**